js/tuxexchange.js

```diff
@@ -0,0 +1,567 @@
+            } else {
+                feeCurrency = this.commonCurrencyCode (this.safeString (trade, 'coin'));
+            }
+            return this.extend (parsedTrade, {
+                'order': this.safeString (trade, 'order'),
+                'fee': {
+                    'cost': this.safeFloat (trade, 'total'),
+                    'currency': feeCurrency,
+                    'rate': this.safeFloat (trade, 'feepercent'),
+                },
+            });
+        } else {
+            return parsedTrade;
+        }
+    }
+
+    async fetchTrades (symbol, since = undefined, limit = undefined, params = {}) {
+        if (symbol === undefined) {
+            throw new ArgumentsRequired (this.id + ' fetchTrades () requires a "symbol" argument');
+        }
+        await this.loadMarkets ();
+        const codes = this.getIdsFromSymbol (symbol);
+        if (codes['base'] !== 'BTC') {
+            throw new NotSupported (this.id + ' this exchange only trades on symbols with BTC as base');
+        }
+        let tradeHistoryRequest = {
+            'method': 'gettradehistory',
+            'coin': this.currencyId (codes.quote),
+            'start': Math.floor (since / 1000),
+            'end': this.seconds (),
+        };
+        const resp = await this.publicGet (tradeHistoryRequest);
+        const trades = this.parseJson (resp);
+        const market = this.getMarket (symbol);
+        const results = this.parseTrades (trades, market, since, limit);
+        return results;
+    }
+
+    async fetchBalance (params = {}) {
+        await this.loadMarkets ();
+        let res = await this.privatePost ({ 'method': 'getmybalances' });
+        const balancesData = this.parseJson (res);
+        const currencies = Object.keys (balancesData);
+        let result = { 'info': balancesData };
+        for (let i = 0; i < currencies.length; i++) {
+            const currency = currencies[i];
+            const balanceData = balancesData[currency];
+            const uppercase = currency.toUpperCase ();
+            const code = this.commonCurrencyCode (uppercase);
+            let account = this.account ();
+            account['total'] = parseFloat (balanceData['balance']);
+            account['used'] = parseFloat (balanceData['frozen']);
+            account['free'] = parseFloat (balanceData['balance'] - balanceData['frozen']);
+            result[code] = account;
+        }
+        return result;
+    }
+
+    async createOrder (symbol, type, side, amount, price = undefined, params = {}) {
+        if (symbol === undefined) {
+            throw new ArgumentsRequired (this.id + ' createOrder () requires a "symbol" argument');
+        }
+        if (price === undefined) {
+            throw new ArgumentsRequired (this.id + ' createOrder () requires a "price" argument');
+        }
+        if (type !== 'limit') {
+            throw new NotSupported (this.id + ' fetchTrades () does not support a "limit" argument for this exchange');
```

Comment:
This exception text is from fetchTrades

Reply:
yep, also we *only* support limit, I have the opposite here. Fixed